torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -1179,6 +1175,36 @@ def call_triton_kernel(
                 "Please use a Config in @triton.autotune instead."
             )
 
+        # Run prune_configs_by to filter the configs
+        if isinstance(variable.kernel, Autotuner):
+            # args and kwargs aren't real values yet
+            # e.g., LazyVariableTracker
+            # If a user wants to access them, we should realize them
+            def realize_var(
+                var: VariableTracker,
+            ) -> Union[Any, VariableTracker, LazyVariableTracker]:
+                while isinstance(var, LazyVariableTracker) and hasattr(var, "realize"):
+                    var = var.realize()
+                # if we can return the value, do it
+                if hasattr(var, "value"):
+                    return var.value
+                elif hasattr(var, "get_real_value"):
+                    return var.get_real_value()
+                else:
+                    return var
+
+            realizedKwargs = {key: realize_var(kwarg) for key, kwarg in kwargs.items()}
+            realizedArgs = zip(
+                variable.kernel.arg_names, [realize_var(arg) for arg in args]
+            )
+            variable.kernel.nargs = dict(realizedArgs)
+
+            variable.kernel.prune_configs(realizedKwargs)
+            # Reset Autotuner vars to the default so we don't run prune_configs again
+            variable.kernel.perf_model = None
+            variable.kernel.configs_top_k = 1.0
+            variable.kernel.early_config_prune = None
+
```

Comment:
Would be helpful (to other folks looking at this code in the future) if the PR body had some more context